addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3678,6 +3688,154 @@ options.registry.WebsiteAnimate = options.Class.extend({
+     */
+    async start() {
+        await this._super(...arguments);
+        this.leftPanelEl = this.$overlay.data("$optionsSection")[0];
+        // Reduce overlay opacity for more highlight visibility on small text.
+        this.$overlay[0].style.opacity = "0.25";
+        this.$overlay[0].querySelectorAll(".o_handle")
+            .forEach(handle => handle.classList.add("pe-none"));
+    },
+    /**
+     * Move "Text Effect" options to the editor's toolbar.
+     *
+     * @override
+     */
+    onFocus() {
+        this.options.wysiwyg.toolbarEl.append(this.$el[0]);
+    },
+    /**
+     * @override
+     */
+    onBlur() {
+        this.leftPanelEl.appendChild(this.el);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Activates & deactivates the text highlight effect.
+     *
+     * @see this.selectClass for parameters
+     */
+    async setTextHighlight(previewMode, widgetValue, params) {
+        return widgetValue ? this._addTextHighlight(widgetValue)
+            : removeTextHighlight(this.$target[0]);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Used to add a highlight SVG element to the targeted text node(s).
+     * This should also take in consideration a situation where many text nodes
+     * are separate e.g. `<p>first text content<br/>second text content...</p>`.
+     * To correctly handle those situations, every set of text nodes will be
+     * wrapped in a `.o_text_highlight_item` that contains its highlight SVG.
+     *
+     * @param {String} highlightID
+     * @private
+     */
+    _addTextHighlight(highlightID) {
+        const highlightEls = [...this.$target[0].querySelectorAll(".o_text_highlight_item svg")];
+        if (highlightEls.length) {
+            // If the text element has a highlight effect, we only need to
+            // change the SVG.
+            highlightEls.forEach(svg => svg.replaceWith(drawTextHighlightSVG(svg.parentElement, highlightID)));
+        } else {
+            applyTextHighlight(this.$target[0], highlightID);
+        }
+    },
+    /**
+     * @override
+     */
+    async _computeWidgetState(methodName, params) {
+        const value = await this._super(...arguments);
+        switch (methodName) {
+            case "selectStyle": {
+                const style = window.getComputedStyle(this.$target[0]);
+                if (value === "currentColor") {
+                    // The highlight default color is the text's "currentColor".
+                    // This value should be handled correctly by the option.
+                    return style.color;
+                } else if (params.cssProperty.startsWith("--text-highlight-width") && !value) {
```

Comment:
why `startsWith` ?

Reply:
If I remember well, it was used for an old version of code :grimacing: => `includes()` used :+1: